user-docs/transaction-functions.md

```diff
@@ -0,0 +1,121 @@
+# Transaction Functions available in `algob`
+
+## `getSuggestedParams`
+
+This function performs the task of fetching the common transaction params from the connected network blockchain. Private chains might throw some error while executing this function, please make sure that the node progresses in that case. 
+
+This function accepts:
++ `algoc1`: an instance(object) of `Algodv2` class of `algosdk`, which is to be used to fetch the transaction params from the connected network. 
+
+This function returns:
++ a promise on resolving which we get the desired transaction params.
+
+The implementation can be found [here](https://github.com/scale-it/algo-builder/blob/master/packages/algob/src/lib/tx.ts#L8).
+
+For more information on Transaction Params, visit [this](https://developer.algorand.org/docs/reference/transactions/).
+
+## `mkTxParams`
+
+This function performs the trivial task of auto-generating the transaction params to be used by the user in creating the transaction object. This basically combines the custom params defined and provided by the user and the suggested params of the blockchain network and returns the combined params in a format which can be easily accepted by the blockchain network. 
+
+This function accepts:
++ `algoc1`: an instance(object) of `Algodv2` class of `algosdk`
++ `userParams`: a dict containing custom params defined by the user
++ `s`: an optional argument which is basically `suggestedParams`, in case this argument isn't provided the function auto-generates the suggested params using the function `getSuggestedParams`.
+
+This function returns:
++ a promise on resolving which we get the desired transaction params.
```

Comment:
what is desired here ?

Reply:
by desired I mean the response user is expecting. I'll change this to something more appropriate.